### PR TITLE
Include API and scripts to get kickstart deployment logs

### PR DESCRIPTION
**Pull request by guioliveirabh**
### Fixes

- [EXDRHUB-970](https://issues.redhat.com/browse/EXDRHUB-970)

### Checklist

- [ ] Related tests were updated
- [ ] Related documentation was updated
- [x] OpenAPI file was updated
- [x] Run `tox`, no tests failed

### CODE REVIEW

File: src/rhub/api/bare_metal/provision.py [hunk 1]
```diff
@@ -84,6 +82,27 @@ def provision_finish(provision_id):
...
+    provision = BareMetalProvision.query.get(provision_id)
+    if not provision:
+        return problem(404, "Not Found", f"Provision {provision_id} does not exist")
+
+    if not isinstance(provision, BareMetalProvisionISO):
+        return problem(403, "Forbidden", f"Provision {provision} does not support log upload at this point")
+
+    file = connexion.request.files["file"]
+    if not file.filename.lower().endswith(".tbz"):
+        return problem(403, "Forbidden", "Provision debug logs needs to be *.tbz")
```
Comment (fpob) [inline src/rhub/api/bare_metal/provision.py:95]: `415 Unsupported Media Type` (RFC 7231) is probably better code for this error.

File: src/rhub/bare_metal/model/data/kickstart_debug_script.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+W cat /proc/partitions
+W mount
+W df -h
+W cat /proc/meminfo
+W cat /proc/cpuinfo
+W ps axf
+W lsof
+W ip -s li
+W ip a
+W ip r
```
Comment (cbouchar) [inline src/rhub/bare_metal/model/data/kickstart_debug_script.sh:31]: If anaconda configured, wondering if we should gather logs that current users come to expect beneath
/var/log/anaconda. Also an important log is console.log Not sure it is included above.  without digging too much into it, I think journalctl will give us this info.

File: src/rhub/bare_metal/model/provision.py [hunk 3]
```diff
@@ -106,19 +108,31 @@ class BareMetalProvisionISO(BareMetalProvision):
...
 
     @property
     def kickstart_file(self) -> Path:
         return BARE_METAL_KICKSTART_BASE_PATH / f"kickstart_{self.id}.cfg"
 
-    def write_kickstart_content(
-        self, kickstart_local_file: TextIO
-    ) -> dict[str, str]:
+    @property
+    def debug_script(self) -> str:
```
Comment (cbouchar) [inline src/rhub/bare_metal/model/provision.py:127]: Really could use some comments here.

**Review by cbouchar — APPROVED** [review]
Looks good to me but I'd wait to make sure Filip is good too.

**Pull request merged**